server/store/utils.js

```diff
@@ -121,45 +121,44 @@ export const createUser = (email, password) => {
 export const createOrGetOneCourse = async (data, option = 'EMPTY') => {
   return new Promise((resolve, reject) => {
     let course = data;
-
     const { school = 'SJSU', code, type } = course;
 
     const splitCourseString = code.split(' ');
     const department = splitCourseString[0];
     const courseCode = splitCourseString[1].replace(/^0+/, '');
 
-    Course.findOne({ school, department, code: courseCode })
+    Course.findOne({
+      school,
+      department,
+      code: courseCode,
+    })
       .then(async foundCourse => {
-        if (foundCourse) return resolve(foundCourse);
-
-        switch (option) {
-          case 'EMPTY':
-            course = {
-              ...data,
-              department,
-              code: courseCode,
-              type,
-            };
-            break;
-          default: {
-            const tasks = [
-              createOrGetAllCourse({
-                codes: data.prerequisites,
-              }),
-              createOrGetAllCourse({
-                codes: data.corequisites,
-              }),
-            ];
-            const [preList, coList] = await Promise.all(tasks);
-            course.prerequisites = preList;
-            course.corequisites = coList;
-            break;
-          }
+        if (foundCourse) {
+          if (type && !foundCourse.type.includes(type))
```

Comment:
Some courses are in multiple types. Meaning they can be in GE and can be a major requirement.